File: src/programmers/p2/P2_42587.java
package programmers.p2;

import java.util.*;

public class P2_42587 {
    public static void main(String[] args) {
        System.out.println(solution(new int[]{2, 1, 3, 2}, 2));
        System.out.println(solution(new int[]{1, 1, 9, 1, 1, 1}, 0));
    }

    public static int solution(int[] priorities, int location) {
        int answer = 0;
        boolean[] list = new boolean[priorities.length];
        list[location] = true;

        List<Integer> queue = new ArrayList<>();

        int length = priorities.length;
        for (int i = 0; i < length; i++) {
            queue.add(i);
        }

        boolean stopFlag = false;
        while (!stopFlag) {
            int index = queue.get(0);
            int currentPriority = priorities[index];
            boolean remove = true;
            for (int priority : priorities) {
                if (currentPriority < priority) {
                    queue.add(queue.get(0));
                    queue.remove(0);
                    remove = false;
                    break;
                }
            }
            if (remove) {
                answer++;
                queue.remove(0);
                priorities[index] = 0;
                stopFlag = list[index];
            }
        }

        return answer;
    }

}

File: src/programmers/p2/P2_42587.kt
package programmers.p2


fun main() {
    println(solution(intArrayOf(2, 1, 3, 2), 2))
    println(solution(intArrayOf(1, 1, 9, 1, 1, 1), 0))
}

private fun solution(priorities: IntArray, location: Int): Int {
    var answer = 0
    var queueList = MutableList(priorities.size) { v -> v }
    var lastIndex = -1
    while (lastIndex != location) {
        if (priorities.max() == priorities[queueList[0]]) {
            priorities[queueList[0]] = -1
            lastIndex = queueList[0]
            answer++
        } else
            queueList.add(queueList[0])
        queueList.removeAt(0)
    }
    return answer
}
